web/src/test/java/com/epam/rft/atsy/web/controllers/rest/CandidateControllerTest.java

```diff
@@ -0,0 +1,284 @@
+    this.mockMvc.perform(get(REQUEST_URL)
+        .param("filter", JSON_CANDIDATE_NAME_AND_EMAIL_AND_PHONE + NON_VALID_FIELD_NAME).param("sort", NAME)
+        .param("order", ASC))
+        .andExpect(status().isOk());
+
+    then(candidateService).should().getAllCandidate(filterRequestArgumentCaptor.capture());
+    assertThat(filterRequestArgumentCaptor.getValue(), equalTo(filterRequest));
+
+    verifyNoMoreInteractions(candidateService);
+  }
+
+  @Test
+  public void loadPageShouldRespondCandidateCollectionWhenOrderByNameAscWithFilterByNameWhenNameIsNotValid()
+      throws Exception {
+
+    final SearchOptions searchOptions =
+        SearchOptions.builder().name(NON_VALID_CANDIDATE_NAME).email(EMPTY_STRING).phone(EMPTY_STRING).build();
+    final FilterRequest filterRequest = FilterRequest.builder().order(SortingRequest.Order.ASC)
+        .fieldName(SortingRequest.Field.NAME).searchOptions(searchOptions).build();
+
+    val filterRequestArgumentCaptor = ArgumentCaptor.forClass(FilterRequest.class);
+    this.mockMvc.perform(get(REQUEST_URL)
+        .param("filter", JSON_NON_VALID_CANDIDATE_NAME).param("sort", NAME).param("order", ASC))
+        .andExpect(status().isOk());
+
+    then(candidateService).should().getAllCandidate(filterRequestArgumentCaptor.capture());
+    assertThat(filterRequestArgumentCaptor.getValue(), equalTo(filterRequest));
+
+    verifyNoMoreInteractions(candidateService);
+  }
+
+  @Test
+  public void loadPageShouldRespondCandidateCollectionWhenOrderByNameAscWithFilterByName() throws Exception {
+    final SearchOptions searchOptions =
+        SearchOptions.builder().name(CANDIDATE_NAME).email(EMPTY_STRING).phone(EMPTY_STRING).build();
+    final FilterRequest filterRequest = FilterRequest.builder().order(SortingRequest.Order.ASC)
+        .fieldName(SortingRequest.Field.NAME).searchOptions(searchOptions).build();
+
+    val filterRequestArgumentCaptor = ArgumentCaptor.forClass(FilterRequest.class);
+    this.mockMvc.perform(get(REQUEST_URL)
+        .param("filter", JSON_CANDIDATE_NAME).param("sort", NAME).param("order", ASC))
+        .andExpect(status().isOk());
+
+    then(candidateService).should().getAllCandidate(filterRequestArgumentCaptor.capture());
+    assertThat(filterRequestArgumentCaptor.getValue(), equalTo(filterRequest));
+
+    verifyNoMoreInteractions(candidateService);
+  }
+
+  @Test
+  public void loadPageShouldRespondCandidateCollectionWhenOrderByNameAscWithFilterByEmailAndPhone() throws Exception {
+    final SearchOptions searchOptions =
+        SearchOptions.builder().name(EMPTY_STRING).email(CANDIDATE_EMAIL).phone(CANDIDATE_PHONE).build();
+    final FilterRequest filterRequest = FilterRequest.builder().order(SortingRequest.Order.ASC)
+        .fieldName(SortingRequest.Field.NAME).searchOptions(searchOptions).build();
+
+    val filterRequestArgumentCaptor = ArgumentCaptor.forClass(FilterRequest.class);
```

Comment:
Please remain consistent and don't use val, if it is not used widely in the project.